src/pages/index.js

```diff
@@ -22,57 +24,117 @@ const profileDescriptionInput = document.querySelector(
   "#profile-description-input"
 );
 
-const editFormValidator = new FormValidator(
-  validationSettings,
-  profileEditForm
-);
-const addFormValidator = new FormValidator(
-  validationSettings,
-  addCardFormElement
-);
-/* --------------------- Create instances of the classes -------------------- */
+// /* -------------------------------------------------------------------------- */
+// /*                                    API;                                    */
+// /* -------------------------------------------------------------------------- */
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "9bd6fe3e-e63d-4619-b9fe-4bda3d127f2c",
+    "Content-Type": "application/json",
+  },
+});
+
+// let cardSection;
+
+// api
+//   .getUserInfo()
+//   .then((data) => {
+//     console.log(data);
+//     userInfo.setUserInfo(data);
+//     userInfo.setAvatar(data.modal);
+//   })
+//   .catch((err) => {
+//     console.log(err);
+//   });
+
+const previewImagePopup = new PopupWithImage("#preview-modal");
+previewImagePopup.setEventListeners();
+
 const cardList = new Section(
   {
-    items: initialCards,
     renderer: (item) => {
       const cardELement = createCard(item);
       cardList.addItem(cardELement);
     },
   },
   ".cards__list"
 );
-cardList.renderItems();
+api
+  .getInitialCards()
+  .then((cards) => {
+    console.log(cards);
+    cardList.renderItems(cards);
+  })
+  .catch((err) => {
+    console.error(err);
+  });
+
+const confirmDeleteModal = new PopupWithConfirm(
+  "#trashcan-modal",
+  handleDeleteClick
+);
+confirmDeleteModal.setEventListeners();
+
+const editAvatarModal = new PopupWithForm(
+  "#edit-avatar-modal",
+  handleAvatarEditSubmit
+);
+editAvatarModal.setEventListeners();
 
 const userInfo = new UserInfo({
   nameSelector: ".profile__title",
   descriptionSelector: ".profile__description",
+  avatarSelector: ".profile__image",
 });
 
 // const addCardModal = new PopupWithForm({
 //   popupSelector: "#add-card-modal",
 //   handleFormSubmit: handleAddCardFormSubmit,
 // });
-const addCardModal = new PopupWithForm(
-  selectors.addCardModalSelector,
-  handleAddCardFormSubmit
-);
+const addCardPopup = new PopupWithForm("#add-card-modal", (data) => {
+  const cardData = {
+    name: data.title,
+    link: data.url,
+  };
 
-addCardModal.setEventListeners();
+  addCardPopup.setLoadingState(true);
+
+  api
+    .createNewCard(cardData)
+    .then((res) => {
+      const card = createCard(res);
+      cardList.addItem(card);
+      addCardFormElement.reset();
+      addFormValidator.resetValidation();
+      addCardPopup.close();
+      addCardPopup.setLoadingState(false);
```

Comment:
`      addCardPopup.setLoadingState(false);
`

Please, delete it . You have it in `finally` (it's enough)